components/script_bindings/codegen/CodegenRust.py

```diff
@@ -1745,13 +1745,16 @@ def __init__(self, descriptor, name, static, unforgeable, crossorigin=False):
                        and (MemberIsLegacyUnforgeable(m, descriptor) == unforgeable or crossorigin)]
         else:
             methods = []
-        self.regular = [{"name": m.identifier.name,
-                         "methodInfo": not m.isStatic(),
-                         "length": methodLength(m),
-                         "flags": "JSPROP_READONLY" if crossorigin else "JSPROP_ENUMERATE",
-                         "condition": PropertyDefiner.getControllingCondition(m, descriptor),
-                         "returnsPromise": m.returnsPromise()}
-                        for m in methods]
+        self.regular = []
+        for m in methods:
+            method = self.methodData(m, descriptor, crossorigin)
+
+            if m.isStatic():
+                method["nativeName"] = CGDictionary.makeMemberName(
+                    descriptor.binaryNameFor(m.identifier.name, True)
+                )
```

Comment:
Why not have methodData() set the nativeName for everything unconditionally?

Reply:
I wanted to change the least that I needed to get this to work. It looks like setting nativeName for everything requires going through and updating all the method declarations and methodinfo declarations to use the new native name as well because those no longer lineup for BinaryName functions. 
I will say this is a bit of hack because it means that any time we have a duplicated method name between static and instance methods we rely on the static method being the one that gets a BinaryName annotation to break the duplication.

Reply:
That's fair. Let's file a followup issue about supporting this more generally so we don't lose track of it?

Reply:
https://github.com/servo/servo/issues/36577